paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -608,6 +608,15 @@ internal class PaymentSheetViewModel @Inject internal constructor(
             val component = DaggerPaymentSheetLauncherComponent
                 .builder()
                 .application(arg.application)
+                .productUsageTokens(
+                    setOfNotNull(
+                        "PaymentSheet",
+                        "deferred-intent".takeIf {
+                            starterArgsSupplier().initializationMode is
+                            PaymentSheet.InitializationMode.DeferredIntent
+                        }
+                    )
+                )
```

Comment:
It seems like we need to pass this in a lot of places, and with different ways to determine the value (`initializationMode` vs. the intent’s client secret). Could we leverage `PaymentMethodCreateParams.productUsage` instead and set it in `IntentConfirmationInterceptor`?